variables.tf

```diff
@@ -64,6 +64,20 @@ variable "kms_encryption_enabled" {
   type        = bool
   description = "Set this to true to control the encryption keys used to encrypt the data that you store for AppID. If set to false, the data is encrypted by using randomly generated keys. For more info on securing data in AppID, see https://cloud.ibm.com/docs/appid?topic=appid-mng-data"
   default     = false
+  validation {
+    condition     = var.kms_encryption_enabled && var.plan != "graduated-tier" ? false : true
+    error_message = "kms encryption is only supported for graduated-tier plan"
+  }
+
+  validation {
+    condition     = !var.kms_encryption_enabled && (var.existing_kms_instance_guid != null || var.kms_key_crn != null) ? false : true
+    error_message = "When passing values for var.existing_kms_instance_guid or/and var.kms_key_crn, you must set var.kms_encryption_enabled to true. Otherwise unset them to use default encryption"
```

Comment:
I think this validation can be removed as it unnecessarily restricts users from using provider-managed keys while having a KMS instance. You can set `kms_encryption_enabled` to false, which will use random (provider-managed keys) with the `existing_kms_instance_guid`, which means you do not need a `kms_key_crn`. 

But if you do need customer-managed keys which means `kms_encryption_enabled` needs to be true, and you must have the `kms_key_crn`. This can be addressed in the below validation.

See the screenshot 
![image](https://github.com/user-attachments/assets/dc34f066-f5a9-4711-a26d-bc3cb3853ee7)

Reply:
I would move the validation to each of the individual inputs. `existing_kms_instance_guid` and `kms_key_crn`